src/main/java/org/spongepowered/common/data/manipulator/immutable/entity/ImmutableSpongeSizeData.java

```diff
@@ -0,0 +1,128 @@
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.manipulator.immutable.entity;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.collect.ComparisonChain;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableSizeData;
+import org.spongepowered.api.data.manipulator.mutable.entity.SizeData;
+import org.spongepowered.api.data.value.immutable.ImmutableBoundedValue;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.common.data.manipulator.immutable.common.AbstractImmutableData;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeSizeData;
+import org.spongepowered.common.data.value.SpongeValueFactory;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+
+public class ImmutableSpongeSizeData extends AbstractImmutableData<ImmutableSizeData, SizeData> implements ImmutableSizeData {
+
+    private final float base;
+    private final float height;
+    private final float scale;
+
+    private final ImmutableBoundedValue<Float> scaleValue;
+
+    public ImmutableSpongeSizeData(float base, float height, float scale) {
+        super(ImmutableSizeData.class);
+        checkArgument(scale > 0);
+        this.scale = scale;
+        this.base = base;
+        this.height = height;
+
+        this.scaleValue = SpongeValueFactory.boundedBuilder(Keys.SCALE)
+                .actualValue(this.scale)
+                .defaultValue(1f)
+                .minimum(0f)
+                .maximum(Float.MAX_VALUE)
+                .build()
+                .asImmutable();
+
+        registerGetters();
+    }
+
+    @Override
+    public ImmutableValue<Float> base() {
+        return new ImmutableSpongeValue<>(Keys.BASE_SIZE, this.base);
+    }
+
+    @Override
+    public ImmutableValue<Float> height() {
+        return new ImmutableSpongeValue<>(Keys.HEIGHT, this.height);
+    }
+
+    @Override
+    public ImmutableBoundedValue<Float> scale() {
+        return this.scaleValue;
+    }
+
+    @Override
+    public SizeData asMutable() {
+        return new SpongeSizeData(this.base, this.height, this.scale);
+    }
+
+    @Override
+    public DataContainer toContainer() {
+        return super.toContainer()
+                .set(Keys.BASE_SIZE.getQuery(), this.base)
+                .set(Keys.HEIGHT.getQuery(), this.height)
+                .set(Keys.SCALE.getQuery(), this.scale);
+    }
+
+    @Override
+    public int compareTo(ImmutableSizeData o) {
+        return ComparisonChain.start()
+                .compare(o.base().get().intValue(), this.base)
+                .compare(o.height().get().intValue(), this.height)
+                .compare(o.scale().get().intValue(), this.scale)
+                .result();
+    }
+
+    public float getBase() {
```

Comment:
Methods not really needed as you can inline it with a lambda below
`registerFieldGetter(Keys.BASE_SIZE, () -> this.base);` etc